doc/src/manual/performance-tips.md

```diff
@@ -90,16 +90,18 @@ On the first call (`@time sum_global()`) the function gets compiled. (If you've
 in this session, it will also compile functions needed for timing.)  You should not take the results
 of this run seriously. For the second run, note that in addition to reporting the time, it also
 indicated that a significant amount of memory was allocated. We are here just computing a sum over all elements in
-a vector of 64-bit floats so there should be no need to allocate memory (at least not on the heap which is what `@time` reports).
+a vector of 64-bit floats so there should be no need to allocate (heap) memory.
+
+We should clarify that what `@time` reports is specifically *heap* allocations, which are typically needed for either mutable objects or for creating/growing variable-sized containers (such as `Array` or `Dict`, strings, or “type-unstable” objects whose type is only known at runtime).  Allocating (or deallocating) such blocks of memory requires an expensive system call (like `malloc` in C), and they must be tracked for garbage collection.  In contrast, immutable values like numbers (except bignums), tuples, and immutable `struct`s can be stored much more cheaply, e.g. in stack or CPU-register memory, with no system call, so one doesn’t typically worry about the performance cost of “allocating” them.
```

Comment:
- Maybe add some line breaks.
- I don't think `malloc` is really a system call
- When you have a GC (or depending on your allocator), a heap allocation can be very cheap (e.g. when allocating into an arena where it is basically just a position bump).
- Perhaps change `“` to `"`?

Reply:
May I suggest edit this directly and merge?

Reply:
How about changing it to *may require an expensive system call (via `malloc` in C)*, which is more technically correct.

I don't want to confuse people with the ins and outs of the GC… the goal here is to give people a more basic sense of what a "heap allocation" is who don't know much about the underlying nuts and bolts.

Reply:
I added line breaks, switched to straight quotes (which are hopefully changed to curly quotes by the documentation system?), added the qualifier "may" before "system call", and noted that this happens "e.g. via" `malloc`.